analysis/behavioral_analysis/demographics_table.ipynb

```diff
@@ -0,0 +1,218 @@
+    "        dict_MM_1year = {f'{sub}_1year':val for (sub,val) in dict_MM_1year.items() if 'MM' in sub}\n",
+    "        \n",
+    "        dict_all = {**dict_HC_baseline, **dict_MM_baseline, **dict_MM_1year}\n",
+    "        df_subs[col_name] = df_subs['sub_ses'].map(dict_all)\n",
+    "\n",
+    "\n",
+    "    df_subs['Employment status'].replace({'self':'Self'},regex=True,inplace=True)\n",
+    "    \n",
+    "    return df_subs\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e1038052",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def save_table1(df_subs,include_pval):\n",
+    "    \n",
+    "    #this includes the THC/CUD items as well\n",
+    "    #columns = ['Sex', 'Age', 'Race', 'Ethnicity', 'Education level', 'Education years','Employment status','Handedness','Positive urine THC at baseline','Positive urine THC at one-year','CUDIT summed score at baseline','CUDIT summed score at one-year','THC frequency per month at baseline','THC frequency per month at one-year']\n",
+    "    #categorical = ['Sex', 'Race', 'Ethnicity', 'Education level','Employment status','Handedness','Positive urine THC at baseline','Positive urine THC at one-year','THC frequency per month at baseline','THC frequency per month at one-year']\n",
+    "    \n",
+    "    #this only includes the demographics\n",
+    "    columns = ['Sex', 'Age', 'Race', 'Ethnicity', 'Education level', 'Education years','Employment status','Handedness']\n",
+    "    categorical = ['Sex', 'Race', 'Ethnicity', 'Education level','Employment status','Handedness']\n",
+    "\n",
+    "    groupby = ['group_ses']\n",
+    "    labels={'HC_baseline': 'HC baseline','MM_baseline': 'MC baseline','MM_1year': 'MC one-year'}\n",
+    "    \n",
+    "    if include_pval:\n",
+    "        mytable = TableOne(df_subs, columns=columns, categorical=categorical, groupby=groupby, rename=labels, pval=True)\n",
+    "    else:\n",
+    "        mytable = TableOne(df_subs, columns=columns, categorical=categorical, groupby=groupby, rename=labels, pval=False)\n",
+    "\n",
+    "    \n",
+    "    #create paths to output dir if not exist\n",
+    "    derivatives_path = '../../../derivatives'\n",
+    "    nilearn_output_path = os.path.join(derivatives_path, 'demographics')\n",
+    "    if not os.path.isdir(nilearn_output_path):\n",
+    "        os.makedirs (nilearn_output_path)\n",
+    "    \n",
+    "    if include_pval:\n",
+    "        mytable.to_csv('../../../derivatives/demographics/table1_all_pval.csv')\n",
+    "        mytable_df = pd.read_csv(f'../../../derivatives/demographics/table1_all_pval.csv')\n",
+    "        mytable_df.drop('Grouped by group_ses.1', axis=1, inplace=True)     \n",
+    "        mytable_df.to_csv('../../../derivatives/demographics/table1_all_pval.csv',index=False)\n",
```

Comment:
same pattern here as before.

Reply:
unless TableOne doesn't return a pandas dataframe or cannot be converted to one.